rrdd/xcp_rrdd.ml

```diff
@@ -475,31 +475,58 @@ let uuid_blacklist = [
 module IntSet = Set.Make(Int)
 
 let domain_snapshot xc =
-  let uuid_of_domain d =
-    Uuid.to_string (Uuid.uuid_of_int_array (d.Xenctrl.handle)) in
+  let dom_uuid_domid_of_domain dom =
```

Comment:
Slightly confusing name. Maybe `uuid_from_domid`? Because of `domid` I think it's clear enough that we are talking about domains.

Reply:
Maybe `metadata_of_domain`? I wanted to capture that we're collecting more than just the uuid.

Reply:
If you want to be brave just `metadata_of` and let the argument speak for itself.